TFNetworkLayer.py

```diff
@@ -2314,6 +2314,7 @@ def __init__(self, activation, **kwargs):
     else:
       self.output_before_activation = OutputWithActivation(x)
     self.output.placeholder = self.output_before_activation.y
+    self.output.dtype = self.output_before_activation.y.dtype.base_dtype.name
```

Comment:
The `__init__` should never change the type/format of `self.output`. This should be done via `get_out_data_from_opts`. The `dtype` (and shape, etc) must be known before any computation graph is actually constructed (via `__init__`).